server/embed/config.go

```diff
@@ -443,15 +443,15 @@ type configJSON struct {
 }
 
 type securityConfig struct {
-	CertFile        string `json:"cert-file"`
-	KeyFile         string `json:"key-file"`
-	ClientCertFile  string `json:"client-cert-file"`
-	ClientKeyFile   string `json:"client-key-file"`
-	CertAuth        bool   `json:"client-cert-auth"`
-	TrustedCAFile   string `json:"trusted-ca-file"`
-	AutoTLS         bool   `json:"auto-tls"`
-	AllowedCN       string `json:"allowed-cn"`
-	AllowedHostname string `json:"allowed-hostname"`
+	CertFile         string   `json:"cert-file"`
+	KeyFile          string   `json:"key-file"`
+	ClientCertFile   string   `json:"client-cert-file"`
+	ClientKeyFile    string   `json:"client-key-file"`
+	CertAuth         bool     `json:"client-cert-auth"`
+	TrustedCAFile    string   `json:"trusted-ca-file"`
+	AutoTLS          bool     `json:"auto-tls"`
+	AllowedCNs       []string `json:"allowed-cns"`
+	AllowedHostnames []string `json:"allowed-hostnames"`
```

Comment:
I am quite surprise they are also changing the json tags upstream since this is a breaking change

In the flag-based config, we updated the go struct keys ( e.g. `AllowedHostnames` ), but kept the old flag name ( e.g. `client-cert-allowed-hostname` ) to make the migration smoother without having to synchronize an update of the binary with an update of the config
https://github.com/DataDog/etcd/blob/54e885aa5cae32c7655a4697773e8ae3fa2a9c5d/server/etcdmain/config.go#L414-L418

No problem with it though this we do not use the file-based config